modules/v2x/v2x_proxy/proto_adapter/proto_adapter.cc

```diff
@@ -0,0 +1,235 @@
+/******************************************************************************
+ * Copyright 2020 The Apollo Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *****************************************************************************/
+
+#include "modules/v2x/v2x_proxy/proto_adapter/proto_adapter.h"
+
+#include <Eigen/Core>
+#include <Eigen/Geometry>
+
+#include "cyber/cyber.h"
+
+namespace v2x {
+namespace proxy {
+
+OSLightColorE AdapterLightColorObu2os(OBULightColorE color) {
+  switch (color) {
+    case OBULightColorE::SingleTrafficLight_Color_RED:
+      return OSLightColorE::SingleTrafficLight_Color_RED;
+    case OBULightColorE::SingleTrafficLight_Color_YELLOW:
+      return OSLightColorE::SingleTrafficLight_Color_YELLOW;
+    case OBULightColorE::SingleTrafficLight_Color_GREEN:
+      return OSLightColorE::SingleTrafficLight_Color_GREEN;
+    case OBULightColorE::SingleTrafficLight_Color_BLACK:
+      return OSLightColorE::SingleTrafficLight_Color_BLACK;
+    case OBULightColorE::SingleTrafficLight_Color_FLASH_GREEN:
+      return OSLightColorE::SingleTrafficLight_Color_FLASH_GREEN;
+    case OBULightColorE::SingleTrafficLight_Color_UNKNOWN:
+    default:
+      return OSLightColorE::SingleTrafficLight_Color_UNKNOWN;
+  }
+}
+
+OSLightTypeE AdapterLightTypeObu2os(int32_t type) {
+  switch (type) {
+    case 2:
+      return OSLightTypeE::SingleTrafficLight_Type_LEFT;
+    case 3:
+      return OSLightTypeE::SingleTrafficLight_Type_RIGHT;
+    case 4:
+      return OSLightTypeE::SingleTrafficLight_Type_U_TURN;
+    case 1:
+    default:
+      return OSLightTypeE::SingleTrafficLight_Type_STRAIGHT;
+  }
+}
+
+bool AdapterLightObu2os(const OBULightT &obu_light,
+                          std::shared_ptr<OSLightT> *os_light_ptr) {
+#if 0
+  RETURN_VAL_IF_NULL(os_light_ptr, false);
```

Comment:
Avoid such usages as `RETURN_IF`, `BREAK_IF`, `CONTINUE_IF`. It hurts readability.